### DISCUSSION TITLE
Move effect bindable properties to different classes

### DISCUSSION THREAD

**Discussion by GalaxiaGuy** [xamarin/XamarinCommunityToolkit#1319] [Ideas]
This thought came about while I was deciding on what the ideal API would be for this feature suggestion:
https://github.com/xamarin/XamarinCommunityToolkit/issues/1280

The existing features define the bindable properties that are used by effects on the shared effect class itself, like here:
https://github.com/xamarin/XamarinCommunityToolkit/blob/main/src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/Semantic/SemanticEffect.shared.cs

This means consuming it through XAML looks like the following:

```xml
<Label xct:SemanticEffect.Description="Blah" />
```

MAUI has things on other classes, for example `SemanticProperties` resulting in XAML like the following:

```xml
<Label SemanticProperties.Description="Blah" />
```

My first approach, (without looking at existing implementations) was to name things entirely for easy of consumption, with XAML like the following:

```xml
<Label ios:Semantic.Description="Blah" />
```

The difference between the things in MAUI and the things in XCT is whether they are implemented in the core, or via an effect. Does the person consuming this care though? The effects generally have logic to auto-attach when you set the property, so having the `Effect` in the property name seems distracting.

Also, it seems that the work in MAUI with handlers might mean that effects are not be needed to implement this kind of platform specific functionality in the future, and not having to update how the features are consumed would be nice.

(From a practical point of view I realise creating new classes with BPs and redirecting and deprecating the existing ones is not great).

**Comment by AndreiMisiukevich** [comment]
As I know, MAUI won't have Effects at all (Just behaviors with option of platform-specific implementation).
I see your point, but please keep in mind that currently any effect can be added as usual (to effects collection).
For example if you want to use events of TouchEffect, you will have to add TouchEffect as regular effect for being possible to subscribe event handlers in xaml (or C# code).

**Reply by GalaxiaGuy** [reply]
Just to be clear, I'm not suggesting any changes to the effects themselves, just where the bindable properties that they use are declared.
